rust/src/smb/log.rs

```diff
@@ -194,162 +195,164 @@ fn smb_common_header(state: &SMBState, tx: &SMBTransaction) -> Json
             if name_raw.len() > 0 {
                 let name = String::from_utf8_lossy(&name_raw);
                 if x.directory {
-                    js.set_string("directory", &name);
+                    jsb.set_string("directory", &name)?;
                 } else {
-                    js.set_string("filename", &name);
+                    jsb.set_string("filename", &name)?;
                 }
             } else {
                 // name suggestion from Bro
-                js.set_string("filename", "<share_root>");
+                jsb.set_string("filename", "<share_root>")?;
             }
             match x.disposition {
-                0 => { js.set_string("disposition", "FILE_SUPERSEDE"); },
-                1 => { js.set_string("disposition", "FILE_OPEN"); },
-                2 => { js.set_string("disposition", "FILE_CREATE"); },
-                3 => { js.set_string("disposition", "FILE_OPEN_IF"); },
-                4 => { js.set_string("disposition", "FILE_OVERWRITE"); },
-                5 => { js.set_string("disposition", "FILE_OVERWRITE_IF"); },
-                _ => { js.set_string("disposition", "UNKNOWN"); },
+                0 => { jsb.set_string("disposition", "FILE_SUPERSEDE")?; },
+                1 => { jsb.set_string("disposition", "FILE_OPEN")?; },
+                2 => { jsb.set_string("disposition", "FILE_CREATE")?; },
+                3 => { jsb.set_string("disposition", "FILE_OPEN_IF")?; },
+                4 => { jsb.set_string("disposition", "FILE_OVERWRITE")?; },
+                5 => { jsb.set_string("disposition", "FILE_OVERWRITE_IF")?; },
+                _ => { jsb.set_string("disposition", "UNKNOWN")?; },
             }
             if x.delete_on_close {
-                js.set_string("access", "delete on close");
+                jsb.set_string("access", "delete on close")?;
             } else {
-                js.set_string("access", "normal");
+                jsb.set_string("access", "normal")?;
             }
 
             // field names inspired by Bro
-            js.set_integer("created", x.create_ts as u64);
-            js.set_integer("accessed", x.last_access_ts as u64);
-            js.set_integer("modified", x.last_write_ts as u64);
-            js.set_integer("changed", x.last_change_ts as u64);
-            js.set_integer("size", x.size);
+            jsb.set_uint("created", x.create_ts as u64)?;
+            jsb.set_uint("accessed", x.last_access_ts as u64)?;
+            jsb.set_uint("modified", x.last_write_ts as u64)?;
+            jsb.set_uint("changed", x.last_change_ts as u64)?;
+            jsb.set_uint("size", x.size)?;
 
             let gs = fuid_to_string(&x.guid);
-            js.set_string("fuid", &gs);
+            jsb.set_string("fuid", &gs)?;
         },
         Some(SMBTransactionTypeData::NEGOTIATE(ref x)) => {
             if x.smb_ver == 1 {
-                let jsa = Json::array();
+                let mut jsa = JsonBuilder::new_array();
```

Comment:
@jasonish does this lead to a mem alloc? If so can we avoid it? Really all we need is to add a `"name":[ ...` and then later a `]`?

Reply:
Yes it would.. Can use open_array here followed by a close.

Reply:
Thanks, Jason!